src/v2/segment.v2.ejs

```diff
@@ -0,0 +1,159 @@
+<%% if (!disable.segment && !sd.THIRD_PARTIES_DISABLED && sd.SEGMENT_WRITE_KEY) { %>
+<script type="text/javascript" defer>
+
+!function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on","addSourceMiddleware","addIntegrationMiddleware","setAnonymousId","addDestinationMiddleware"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t,e){var n=document.createElement("script");n.type="text/javascript";n.async=!0;n.src="https://cdn.segment.com/analytics.js/v1/"+t+"/analytics.min.js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(n,a);analytics._loadOptions=e};analytics.SNIPPET_VERSION="4.1.0";
+
+// adaptation of:
+//   https://gist.github.com/sperand-io/4725e248a35d5005d68d810d8a8f7b29
+//   https://github.com/littledata/shopify-tracker/blob/ec4fac54cd56344980aa481800ee8dba550ea97d/src/customerPrivacyAPI/oneTrust.ts#L31
+
+const SEGMENT_WRITE_KEY = "<%%= sd.SEGMENT_WRITE_KEY %>";
+const SEGMENT_MAPPING = 'C0001'; // Segment itself assigned OneTrust Strictly Necessary category.
+let SEGMENT_DESTINATIONS = [];
+
+fetchDestinations(SEGMENT_WRITE_KEY).then(
+  destinations => {
+    SEGMENT_DESTINATIONS = destinations;
+
+    waitForOneTrust();
+
+    if (oneTrustReady()) {
+      getConsentAndLoadSegment();
+    }
+  }
+)
+
+// OneTrust calls OptanonWrapper when there's a consent change.
+window.OptanonWrapper = getConsentAndLoadSegment;
```

Comment:
Observation: This is always triggered once even when there's no consent change.

Reply:
After testing it a bunch of times, it looks like OneTrust triggers `OptanonWrapper` once every page load, even when there's no consent change. No idea why the trigger and when. I thought that it happens when OneTrust has successfully set `window.OnetrustActiveGroups`, but that's likely not the case, because the trigger happens on a page reload (`window.OnetrustActiveGroups` survives page reloads). I wonder if there's potential for a reload loop. Scenario: Page reloads, script fires, loads Segment, registers `OptanonWrapper`, a little later OneTrust triggers `OptanonWrapper`, script sees that Segment has already been initialized, script reloads page, repeat. To guard against that, I am having `getConsentAndLoadSegment` abort if there's not really a consent change, which requires saving OneTrust consent into a var.